[ADD] accounting: electronic invoicing (edi) (#2231)
odoo/documentation: content/applications/finance/accounting/receivables/customer_invoices/electronic_invoicing.rst

====================
Electronic invoicing

> ====================

Author: Are abbreviations working in titles? I didn't manage to apply it

> ====================

Reviewer: Yes it does :) @Donapi 
It didn't work even with copy-paste?

EDI, or electronic data interchange, allows to rapidly send information digitally using standard

> formats.

Reviewer: longer explanation maybe, to explain what it means. If I don't know EDI yet, I can't understand from your intro that it means the invoices data is included.

This feature enables automating the administration between companies and might also be required by
some governments for fiscal control or to facilitate the administration.

> 

Reviewer: Add an explanation that there are various formats (it is still an explanation, so no need to link this info to what Odoo does).

Any business that processes large volumes of documents and forms can benefit from an EDI solution to
increase automation and improve accuracy.

.. seealso::
- :ref:`Odoo and EDI <https://www.odoo.com/blog/odoo-news-5/odoo-and-edi-766#blog_content>`
- :ref:`Fiscal localization packages <overview/fiscal-localization-package>`
- :ref:`List of supported countries <overview/localizations-list>`

Configuration
=============

Go to :menuselection:`Accounting --> Configuration --> Journals --> Customer Invoices -->

> Advanced Settings --> Electronic Invoicing` and select the boxes you need.

Reviewer 2: I'd go with "enable" for the word "boxes", I feel it is more colloquial. Or you could say "select the _options_ you need"
```suggestion
Advanced Settings --> Electronic Invoicing` and select the boxes you need.
```

> Advanced Settings --> Electronic Invoicing` and check the boxes you need.

Reviewer: instead of "check boxes" maybe "select the formats you want to enable for this journal" or something like that?
+ small screenshot can't hurt here.

> Once :guilabel:`Electronic Invoicing` is selected,  XML documents are generated, either visible or

Reviewer: Once an electronic invoicing is enabled,

embedded in the PDF.

.. note::

>    - By default, the :guilabel:`Factur-X` option is checked. It means that an XML file is automatically

Reviewer: checked --> enabled?

     included in the PDF document that is sent.

>    - The options available depend on your localization.

Reviewer: on your company's country (+ as defined in your settings?)

>    - Odoo supports the Peppol format bis3 that can be used via existing access points. Odoo should
>      soon become an access point itself.

Reviewer: ```suggestion
   - Odoo supports the Peppol format bis3 that can be used via existing access points.
```
announcements are rather for blogs, not really for doc pages.

================================================================
Electronic invoicing (:abbr:`EDI (electronic data interchange)`)
================================================================

EDI, or electronic data interchange, is the inter-company communication of business documents, such
as purchase orders and invoices, in a standard format. Sending documents according to an EDI
standard ensures that the machine receiving the message can interpret the information correctly.

> Various EDI file formats exist.

Reviewer 3: Is it various formats according to countries? If so maybe it's nice to specify it, at this point of the page.

EDI feature enables automating the administration between companies and might also be required by
some governments for fiscal control or to facilitate the administration.

Any business that processes large volumes of documents and forms can benefit from an EDI solution to
increase automation and improve accuracy.

Since July 2022, Odoo supports, among others, the following formats.

.. list-table::
   :header-rows: 0

   * - Factur-X (CII)
     - Default format on Odoo (enabled by default)

>    * - Peppol BIS Billing 3.0
>      - For companies which countries are part of the `EAS list
>        <https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/>`_
>    * - E-FFF
>      - For Belgian companies
>    * - NLCIUS
>      - For Dutch companies
>    * - XRechnung UBL
>      - For German companies

Reviewer: ```suggestion
   * - Peppol BIS Billing 3.0 (UBL)
     - For companies which countries are part of the `EAS list
       <https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/>`_
   * - E-FFF
     - For Belgian companies
   * - NLCIUS
     - For Dutch companies
   * - XRechnung (UBL)
     - For German companies
```

   * - EHF 3.0

>      - Norwegian companies have to use Peppol BIS Billing 3.0 that comply with the EHF 3.0 requirements

Reviewer 4: ```suggestion
     - Norwegian companies have to use Peppol BIS Billing 3.0 which complies with the EHF 3.0 requirements
```
Maybe we can add the link to the norwegian authority explaining the what and why ? https://anskaffelser.dev/postaward/g3/spec/current/billing-3.0/norway/#_implementation

.. seealso::
   - :doc:`../../fiscal_localizations/overview/fiscal_localization_packages`
   - :doc:`../../fiscal_localizations/overview/localizations_list`
   - `Odoo and EDI <https://www.odoo.com/blog/odoo-news-5/odoo-and-edi-766#blog_content>`_

Configuration
=============

Go to :menuselection:`Accounting --> Configuration --> Journals --> Customer Invoices -->
Advanced Settings --> Electronic Invoicing` and enable the formats you need for this journal.

> .. image:: electronic_invoicing/formats.png

Reviewer 4: I like the image, I would keep it, but there's a task to hide the Factur-X setting on the journal...
SInce it's the inter operability format for Odoo, we don't want our users to unselect this format. In addition, it's the french EDI formats so most of our customers don't care about it and will wonder why this format is appearing on their journal. The task is here: https://www.odoo.com/web#id=2894072&cids=1&menu_id=4720&action=333&active_id=967&model=project.task&view_type=form
If it's reviewed and deployed in a close future (and only for the future v16), we can change the image later :) .

   :align: center

>    :alt: Select the EDI format you need

Reviewer 3: Since you only get the option based on your country, by seeing all the options here on the picture, I kind of get confused and wonder if I should have the same or not. 

By reading the explanation right before the picture all is clear, the picture IMO can be misleading.

Once an electronic invoicing format is enabled, XML documents are generated, either visible or

> embedded in the PDF.

Reviewer 4: FYI (maybe it can be useful to mention it)

the EDI documents are generated when clicking on the 'Confirm' button on top of the invoices/credit notes. 

The PDF are generated when clicking 'Send & Print' or 'Print' buttons. As E-FFF requires the PDF to be embedded in the xml (yes, yes, the PDF is inside the xml), it will only appear in the attachments when clicking on the 'Print' or 'Send & Print' button (after having confirmed the invoice).

.. note::
   - By default, the :guilabel:`Factur-X` option is enabled. It means that an XML file is
     automatically included in the PDF document that is sent.

>    - The formats available depend on your company's localization.

Reviewer: localization... our country (in the address form)?

>    - Odoo supports the **Peppol format bis3** that can be used via existing access points.

Reviewer: Check the correct ppellation for the peppol format.
Maybe "Peppol BIS format"?

>    - Odoo supports the **Peppol format bis3** that can be used via existing access points.

Reviewer 4: ```suggestion
   - Odoo supports the **Peppol BIS Billing 3.0** that can be used via existing access points.
```

.. important::
   If your database was created before July 2022 and you wish to use one of the new formats, you

>    have to install the module `account_edi_ubl_cii` and your country-specific module.

Reviewer 3: Here for module you put '..' while just below in the example you use (..) + simple characters, maybe unify since it's in the same page

.. example::
   If you want to use the Belgian format E-FFF in a database created prior to July 2022, you need to

>    install the Belgian module (l10n_be_edi) and the account_edi_ubl_cii.

Reviewer 3: Same as before, maybe it's fine to use '..' for modules

.. note::

>    Once the new module is installed, UBL 2.0 and UBL 2.1 formats no longer appear in the journal.

Reviewer: - "new" is relative and deprecates really quickly. writing "new format" isn't recommended.
- writing the technical names of modules isn't recommended, as users generally don't work in developer mode. See for example https://www.odoo.com/documentation/15.0/applications/finance/accounting/fiscal_localizations/localizations/luxembourg.html
- if the example and note are valid for the text in "important", you could try to ident them to make them part of the important. But it's already a lot of content on its own. This topic is maybe worth a heading of its own? Up to you.

Author: Updated. I tried several options and kept the better one according to me.

>    Once the new module is installed, UBL 2.0 and UBL 2.1 formats no longer appear in the journal.

Reviewer 4: ```suggestion
   Once the new module is installed, UBL 2.0 and UBL 2.1 formats no longer appear in the journal.
```
Maybe mention why we chose not to show them: it seemed no longer useful since they can use the more widely known UBL BIS Billing 3.0 (anyway, in most cases, there are few differences between these formats)

================================================================
Electronic invoicing (:abbr:`EDI (electronic data interchange)`)
================================================================

EDI, or electronic data interchange, is the inter-company communication of business documents, such
as purchase orders and invoices, in a standard format. Sending documents according to an EDI
standard ensures that the machine receiving the message can interpret the information correctly.
Various EDI file formats exist and are available depending on your company's country.

EDI feature enables automating the administration between companies and might also be required by
some governments for fiscal control or to facilitate the administration.

Any business that processes large volumes of documents and forms can benefit from an EDI solution to
increase automation and improve accuracy.

> Odoo supports, among others, the following formats.

Reviewer: Isn't the list longer, with the info you got from Julien?

.. list-table::
   :header-rows: 1

   * - Format Name
     - Applicability
   * - Factur-X (CII)
     - Default format on Odoo (enabled by default)
   * - Peppol BIS Billing 3.0 (UBL)
     - For companies which countries are part of the `EAS list
       <https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/>`_
   * - E-FFF
     - For Belgian companies
   * - XRechnung (UBL)
     - For German companies
   * - Fattura PA (IT)
     - For Italian companies
   * - CFDI (4.0)
     - For Mexican companies
   * -  Peru UBL 2.1
     - For Peruvian companies
   * - SII IVA Llevanza de libros registro (ES)
     - For Spanish companies

.. seealso::
   - :doc:`../../fiscal_localizations/overview/fiscal_localization_packages`

>    - :doc:`../../fiscal_localizations/overview/localizations_list`

Reviewer: If the list above isn't more detailed, I think this link may be misleading, since it may suggest we're talking about EDI supported according to the countries listed. It isn't a great page anyway and I wouldn't promote it.

Author: I removed the links then

Go to :menuselection:`Accounting --> Configuration --> Journals --> Customer Invoices -->
Advanced Settings --> Electronic Invoicing` and enable the formats you need for this journal.

.. image:: electronic_invoicing/formats.png
   :align: center
   :alt: Select the EDI format you need

Once an electronic invoicing format is enabled, XML documents are generated when clicking on
:guilabel:`Confirm` in documents such as invoices, credit notes, etc. These documents are either
visible in the attachment section, or embedded in the PDF.

.. note::
   - For E-FFF, the xml file only appears after having generated the PDF (:guilabel:`Print` or
     :guilabel:`Send & Print` button), since the PDF needs to be embedded inside the xml.
   - By default, the :guilabel:`Factur-X` option is enabled. It means that an XML file is
     automatically included in the PDF document that is sent.
   - The formats available depend on the country registered in your company's :guilabel:`General
     Information`.
   - Odoo supports the **Peppol BIS Billing 3.0** format that can be used via existing access
     points.

Adding formats available from July 2022 in a database created prior to July 2022

> --------------------------------------------------------------------------------

Reviewer: Maybe shorter?
Add new formats to a database created before July 2022

If your database was created before July 2022 and you wish to use one of the formats available from
July 2022, you have to install the module :guilabel:`Import/Export electronic invoices with UBL/CII`
(module’s technical name: `account_edi_ubl_cii`) and your country-specific module.

.. example::
   If you want to use the Belgian format E-FFF in a database created prior to July 2022, you need to

>    install the Belgian module `l10n_be_edi` and the `account_edi_ubl_cii`.

Reviewer: write the modules' name like you did above (normal name + technical name). You could even put them in a very short list like we did in other localization pages.

> If your database was created before July 2022 and you wish to use one of the formats available from
> July 2022, you have to install the module :guilabel:`Import/Export electronic invoices with UBL/CII`
> (module’s technical name: `account_edi_ubl_cii`) and your country-specific module.
> 
> .. example::
>    If you want to use the Belgian format E-FFF in a database created prior to July 2022, you need to
>    install the Belgian module `l10n_be_edi` and the `account_edi_ubl_cii`.

Reviewer: ```suggestion
If your database was created before July 2022 and you wish to use one of the formats available from
July 2022, you have to :ref:`install <general/install>` the module :guilabel:`Import/Export 
electronic invoices with UBL/CII` (module’s technical name: `account_edi_ubl_cii`) and your 
country-specific module.

.. example::
   If you want to use the Belgian format E-FFF in a database created prior to July 2022, you need to
   :ref:`install <general/install>`:

   .. list-table::
      :header-rows: 1

      * - Name
        - Technical name
      * - Belgium - E-Invoicing (UBL 2.0, e-fff)
        - l10n_be_edi
      * - Import/Export electronic invoices with UBL/CII
        - account_edi_ubl_cii
```

================================================================
Electronic invoicing (:abbr:`EDI (electronic data interchange)`)
================================================================

EDI, or electronic data interchange, is the inter-company communication of business documents, such
as purchase orders and invoices, in a standard format. Sending documents according to an EDI
standard ensures that the machine receiving the message can interpret the information correctly.
Various EDI file formats exist and are available depending on your company's country.

EDI feature enables automating the administration between companies and might also be required by
some governments for fiscal control or to facilitate the administration.

Electronic invoicing of your documents such as customer invoices, credit notes or vendor bills is
one of the application of EDI.

Odoo supports, among others, the following formats.

.. list-table::
   :header-rows: 1

   * - Format Name
     - Applicability
   * - Factur-X (CII)
     - Default format on Odoo (enabled by default)
   * - Peppol BIS Billing 3.0 (UBL)

>      - For companies which countries are part of the `EAS list

Reviewer 5: ```suggestion
     - For companies whose countries are part of the `EAS list
```

     :guilabel:`Send & Print` button), since the PDF needs to be embedded inside the xml.
   - By default, the :guilabel:`Factur-X` option is enabled. It means that an XML file is
     automatically included in the PDF document that is sent.
   - The formats available depend on the country registered in your company's :guilabel:`General
     Information`.
   - Odoo supports the **Peppol BIS Billing 3.0** format that can be used via existing access
     points.

Add new formats to a database created before July 2022
------------------------------------------------------

If your database was created before July 2022 and you wish to use one of the formats available from
July 2022, you have to :ref:`install <general/install>` the module :guilabel:`Import/Export
electronic invoices with UBL/CII` (module’s technical name: `account_edi_ubl_cii`) and your
country-specific module.

.. example::
   If you want to use the Belgian format E-FFF in a database created prior to July 2022, you need to
   :ref:`install <general/install>`:

   .. list-table::
      :header-rows: 1

      * - Name
        - Technical name
      * - Belgium - E-Invoicing (UBL 2.0, e-fff)

>         - l10n_be_edi
>       * - Import/Export electronic invoices with UBL/CII
>         - account_edi_ubl_cii

Reviewer 5: ```suggestion
        - `l10n_be_edi`
      * - Import/Export electronic invoices with UBL/CII
        - `account_edi_ubl_cii`
```
